### DISCUSSION TITLE
v1.0.0 🎉

### DISCUSSION THREAD

**Discussion by vobu** [ui5-community/wdi5#377] [Announcements]
# v1.0.0
Here we are, with the first major release of `wdi5`!
It includes a plethora of features and changes. Let's dive in 🤿 

## 🔐 Authentication support
Long in the making, now here: configure `wdi5` to authenticate against 
- BTP (default IdP)
- Office 365
- a custom IdP 
- and with Basic Auth

📖 https://ui5-community.github.io/wdi5/#/authentication
Authentication is config-only, so no changes in test code required 😎 

## 🤲 UI5 Managed Object API alignment
In addition to the UI5 Control API alignment, now the managed objects in `wdi5`-/Node.js-scope are also aligned:
```js
// assuming title is a sap.m.Title
const title = await browser.asControl(selector)
// bindingInfo is a UI5 Managed Object, not a UI5 control!
const bindingInfo = await title.getModel() // same .getModel in wdi5 and UI5!
const name = await bindingInfo.getProperty("/path/in/model") // same .getProperty in wdi5 and UI5!
```
more on this in https://ui5-community.github.io/wdi5/#/usage 

## 🗑️ deprecation of the `wdi5`-specific `url` configuration option
the always mandatory `basicUrl` and the `wdi5`-specific `url` was confusing, so it now is `baseUrl` only
→ less config clutter
📖 https://ui5-community.github.io/wdi5/#/configuration?id=url-deprecated
♻️ no worries, legacy support is in place, so all your configs keep on working

## 📝 documentation enhancements
- more [recipes](https://ui5-community.github.io/wdi5/#/recipes), now sorted alphabetically
- [@wdi5 fosstodon](https://fosstodon.org/@_wdi5_) account
- Journey- and Test-Recorder [→ tools](https://ui5-community.github.io/wdi5/#/resources?id=tools)

💑 Thanks a ton to all contributors, testers and supporters all over [UI5-land](https://github.com/ui5-community/)!
**This one's for you! 👏 ✌️** 

PS: expect next releases to follow a more proper `major`-`minor`-`patch` semantic versioning strategy. So far, we stuck to `0.9.x` patch releases in order to prep for this `1.0`, which is now done. So true semantic versioning it will be from now on.

**Comment by akp3210** [comment]
@vobu How do we pass authentication details such as username and password?

**Comment by monavari-lebrecht** [comment]
you can do this by providing it as an ENVIRONMENT variable. E.g. you can create a  file named ".env" within your working directory with the contents:

wdi5_username=username@domain.de
wdi5_password=myPassPhrase

For MultiRemote Access add the name of the remote.

You can find further information here: https://ui5-community.github.io/wdi5/#/authentication?id=credentials
